servers/zms/src/main/java/com/yahoo/athenz/zms/store/impl/jdbc/JDBCConnection.java

```diff
@@ -1081,10 +1081,15 @@ public boolean insertDomainTags(String domainName, Map<String, StringList> tags)
             throw notFoundError(caller, ZMSConsts.OBJECT_DOMAIN, domainName);
         }
         int curTagCount = getDomainTagsCount(domainId);
-        int remainingTagsToInsert = domainTagsLimit - curTagCount;
+        int newTagCount = calculateTagCount(tags);
+        if (curTagCount + newTagCount > domainTagsLimit) {
```

Comment:
why are we adding the current tag count to new tag count ?

Reply:
For tag updates, we delete (if needed) then insert.
While calculating the quota, we should consider existing tags.

Reply:
but this will be a put meta call which will have the updated list of tags (including the current tags or deleted tags)

Reply:
👍